index.js

```diff
@@ -26,18 +27,37 @@ const stripCookieHeader = header => {
 	return header.substring(start, end + 1);
 };
 
+const parseCookieName = cookieName => {
+	const nameMatch = COOKIE_NAME.exec(cookieName);
+	if (nameMatch === null) {
+		return null;
+	}
+	return nameMatch[1];
+};
+
+const parseCookieValue = cookieValue => {
+	const valueMatch = COOKIE_VALUE.exec(cookieValue);
+	if (valueMatch === null) {
+		return null;
+	}
+	return valueMatch[1] === undefined ?
+		valueMatch[2] :
+		valueMatch[1];
+};
+
 const parseCookiePair = cookiePair => {
-	const match = COOKIE_PAIR.exec(cookiePair);
+	const parts = cookiePair.split('=', 2);
```

Comment:
I can’t check right now, but I think this will allow invalid cookie pairs to be parsed when they contain more than one `=` because of the unusual way `split` works in JavaScript. (Also, `=` is valid in quoted values.) Try splitting manually with `indexOf`?

Reply:
Hmm, if I'm reading this right, a valid `cookieName` can't contain an `=`, so this should always split a cookie pair into name and value. 

Do you see an edge case if a user provides some pathological input? I thought `parseCookiePair('name"="foo"')` might not work, but I added a test for it, just returns `null`. I'm kinda struggling to come up with a case where this doesn't work as expected.

This is the same as doing 
```js
const i = cookiePair.indexOf('='); 
const part0 = cookiePair.slice(0, i); 
const part1 = cookiePair.slice(i);
```
, just less verbose, I think.

Reply:
`.split('=', 2)` is like `.split('=').slice(0, 2)`, so I expect the problematic cases to be `a=b=c` (shouldn’t parse) and `a="b=c"` (should parse).

Reply:
Oh... wow... I never realized that JS continues to split past the `limit` parameter... :grimacing: 

Okay, fixed that. `a="b=c"` parses as expected. `a=b=c` parses to the same value as `a="b=c"`, which I believe is correct, since the unquoted cookie value is allowed to contain equals signs (according to my understanding). Pushed up some unit tests for these cases, lmk if that looks good.

Reply:
Oh, you’re right, `a=b=c` should parse. Are you pushing the changes to the right place? The pull request doesn’t seem to have updated.

Reply:
Been pushing to the wrong branch, check now.